app/PaymentDrivers/Authorize/ChargePaymentProfile.php

```diff
@@ -44,19 +49,42 @@ public function chargeCustomerProfile($profile_id, $payment_profile_id, $amount)
         $profileToCharge->setPaymentProfile($paymentProfile);
 
         $invoice_numbers = '';
-
-        if($this->authorize->payment_hash->data)
-            $invoice_numbers =  collect($this->authorize->payment_hash->data->invoices)->pluck('invoice_number')->implode(',');
+        $taxAmount = 0;
+        $invoiceTotal = 0;
+        $invoiceTaxes = 0;
+
+        if($this->authorize->payment_hash->data) {
+            $invoice_numbers =  collect($this->authorize->payment_hash->data->invoices)->pluck('invoice_number')->implode(",");
+            $invObj = Invoice::whereIn('id', $this->transformKeys(array_column($this->authorize->payment_hash->invoices(), 'invoice_id')))
+                ->withTrashed()->get()->pluck('amount', 'total_taxes')->all();
+            
+            foreach($invObj as $invTaxes=>$invAmount) {
+                $invoiceTotal += $invAmount;
+                $invoiceTaxes += $invTaxes;
+            }
+            if ($invoiceTotal != $amount) {
```

Comment:
you may get burnt with a comparison operator issue here.

it may be prudent to use something like

round($invoiceTotal,2) != round($amount,2)

to ensure comparisons will match

Reply:
$amount is already round